packages/rfw/lib/src/dart/text.dart

```diff
@@ -2327,16 +2403,39 @@ class _Parser {
         _advance();
         return _withSourceRange(LoopReference(_loopIdentifiers.length - index, _readParts(optional: true)), start);
       }
-      return _readConstructorCall();
+      return _readConstructorCall(widgetBuilderScope: widgetBuilderScope);
     }
     throw ParserException._unexpected(_source.current);
   }
 
-  ConstructorCall _readConstructorCall() {
+  WidgetBuilderDeclaration _readWidgetBuilderDeclaration({
+    List<String> widgetBuilderScope = const <String>[],
+  }) {
+    _expectSymbol(_SymbolToken.openParen);
+    final _Token argumentNameToken = _source.current;
+    final String argumentName = _readIdentifier();
+    _checkIsNotReservedWord(argumentName, argumentNameToken);
+    _expectSymbol(_SymbolToken.closeParen);
+    _expectSymbol(_SymbolToken.equals);
+    _expectSymbol(_SymbolToken.greatherThan);
+    final Object widget = _readValue(
+      extended: true,
+      widgetBuilderScope: <String>[...widgetBuilderScope, argumentName],
+    );
+    assert(widget is ConstructorCall || widget is Switch);
```

Comment:
as in the binary section, this seems like it should be an exception, not an assert. Asserts are for things that, if they occur, indicate an error in the code, but here you're using it to catch an error in the data being parsed, unless i misunderstand something.

Reply:
Great call. Fixed!